### PR TITLE
magento/magento2#: GraphQl. DeletePaymentToken. Remove redundant validation logic. Test coverage.

### PR DESCRIPTION
**Pull request by atwixfirster**
### Description (*)

Pull request:
- removes a redundant validation logic from `\Magento\VaultGraphQl\Model\Resolver\DeletePaymentToken` because `Webonix` will cover a removed validation logic.
- adds a unit test for mentioned class

### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

Thank you!

### CODE REVIEW

File: app/code/Magento/VaultGraphQl/Test/Unit/Model/Resolver/DeletePaymentTokenTest.php [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    /**
+     * @var PaymentTokenRepositoryInterface|MockObject
+     */
+    private $paymentTokenRepositoryMock;
+
+    /**
+     * @var PaymentTokenInterface|MockObject
+     */
+    private $paymentTokenMock;
+
```
Comment (rogyar) [inline app/code/Magento/VaultGraphQl/Test/Unit/Model/Resolver/DeletePaymentTokenTest.php:70]: Please, make sure you don't have duplicated empty lines.

File: app/code/Magento/VaultGraphQl/Test/Unit/Model/Resolver/DeletePaymentTokenTest.php [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+            ->method('getByPublicHash')
+            ->willReturn($this->paymentTokenMock);
+
+        $this->paymentTokenRepositoryMock
+            ->expects($this->once())
+            ->method('delete')
+            ->with($this->paymentTokenMock)
+            ->willReturn($paymentTokenResult);
+
+        $this->resolver->resolve(
```
Comment (rogyar) [inline app/code/Magento/VaultGraphQl/Test/Unit/Model/Resolver/DeletePaymentTokenTest.php]: I would recommend using assertions (`self::assert`) when a testable method is supposed to return something.

**Review by rogyar — CHANGES_REQUESTED** [review]
Hi @atwixfirster. Thank you for providing an additional unit test, that's very useful. However, the provided unit test might not exactly cover the modification provided in this PR.

I would suggest extending the [following](https://github.com/magento/magento2/blob/288dec36c94dc7372ccb540095434cab815607f1/dev/tests/api-functional/testsuite/Magento/GraphQl/Vault/CustomerPaymentTokensTest.php#L20) API-functional test with a simple `testDeletePaymentTokenWithNoPublicHash` test. Since there are similar tests in the mentioned class, it should not be a big deal.

Thank you!

File: app/code/Magento/VaultGraphQl/Test/Unit/Model/Resolver/DeletePaymentTokenTest.php (hunk 1 above)
Comment (atwixfirster) [reply app/code/Magento/VaultGraphQl/Test/Unit/Model/Resolver/DeletePaymentTokenTest.php:70]: Yep... **fixed**

File: app/code/Magento/VaultGraphQl/Test/Unit/Model/Resolver/DeletePaymentTokenTest.php (hunk 2 above)
Comment (atwixfirster) [reply app/code/Magento/VaultGraphQl/Test/Unit/Model/Resolver/DeletePaymentTokenTest.php]: Good notice!

**implemented**

Thanks @rogyar

**Comment by atwixfirster** [comment]
> Hi @atwixfirster. Thank you for providing an additional unit test, that's very useful. However, the provided unit test might not exactly cover the modification provided in this PR.
> 
> I would suggest extending the [following](https://github.com/magento/magento2/blob/288dec36c94dc7372ccb540095434cab815607f1/dev/tests/api-functional/testsuite/Magento/GraphQl/Vault/CustomerPaymentTokensTest.php#L20) API-functional test with a simple `testDeletePaymentTokenWithNoPublicHash` test. Since there are similar tests in the mentioned class, it should not be a big deal.
> 
> Thank you!

Regarding to @lenaorobei notice, we do not cover framework (`Webonix`) errors in the API-unctional test:



Thank you!

**Comment by rogyar** [comment]
Probably, that makes sense. If we would cover every Webonyx validation case, we could have a really huge test case. 

Anyway, thank you for the fix

**Review by rogyar — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-6636](https://jira.corp.magento.com/browse/ENGCOM-6636) has been created to process this Pull Request

**Comment by engcom-Bravo** [comment]
:heavy_check_mark:  QA Passed

Hello @atwixfirster 

- The PR adds a new Unit test and the test Passes.
- Also, the redundant validation logic is removed from ```\Magento\VaultGraphQl\Model\Resolver\DeletePaymentToken```

**Pull request merged**